docs/guides/data-ingestion.mdx

```diff
@@ -0,0 +1,196 @@
+
+In case of little size records, ReductStore provides a batch ingestion mode. In this mode, the server receives a batch of records in a single request to reduce the overhead of the HTTP protocol.
+
+
+### Limitations
+
+The following limitations are applied to the data ingestion process:
+
+* Currently, records and their metadata are immutable. Once a record is ingested successfully, it cannot be overwritten or changed.
+* ReductStore doesn't have any limitations on the size of the record, however, the metadata is sent in the HTTP headers of the request and the size of the headers can be limited by the server or client configuration.
+* Since ReaductStore pre-allocate space for records and store them together in a block, the size must be known in advance even though the record is sent in chunks.
+
+## Typycal Data Ingestion Cases
+
+In this section, you can learn how to implement typical data ingestion cases using the ReductStore SDKs or HTTP API. Please note that all examples have been created for a local ReductStore instance accessible at **http://127.0.0.1:8383** using the API token 'my-token'.
+
+For more information on how to set up a local ReductStore instance, refer to the [**Getting Started**](../getting-started) guide.
+
+
+### Simple Data Ingestion
+
+The simplest way to write data into ReductStore is by sending an entire single record. In this case, the record is transmitted as a binary object in the body of the POST request.
+
+import SimplePy from "!!raw-loader!./examples/py/src/data_ingestion_simple.py";
+import SimpleJs from "!!raw-loader!./examples/js/src/data_ingestion_simple.mjs";
+import SimpleRs from "!!raw-loader!./examples/rs/examples/data_ingestion_simple.rs";
+import SimpleCpp from "!!raw-loader!./examples/cpp/src/data_ingestion_simple.cc";
+import SimpleCurl from "!!raw-loader!./examples/curl/data_ingestion_simple.sh";
+
+<Tabs>
+    <TabItem value="Python">
+        <CodeBlock language="python">
+            {SimplePy}
+        </CodeBlock>
+    </TabItem>
+    <TabItem value="JavaScript">
+        <CodeBlock language="javascript">
+            {SimpleJs}
+        </CodeBlock>
+    </TabItem>
+    <TabItem value="Rust">
+        <CodeBlock language="rust">
+            {SimpleRs}
+        </CodeBlock>
+    </TabItem>
+    <TabItem value="C++">
+        <CodeBlock language="cpp">
+            {SimpleCpp}
+        </CodeBlock>
+    </TabItem>
+    <TabItem value="cURL">
+        <CodeBlock language="bash">
+            {SimpleCurl}
+        </CodeBlock>
+    </TabItem>
+</Tabs>
+
+### Streaming Data
+
+For large records, using the streaming approach is recommended. Here, a client application sends a record in chunks, avoiding loading the entire record into memory. This approach is particularly beneficial for real-time data ingestion, where data is received from a stream and needs to be sent to ReductStore without buffering.
+
+:::info
```

Comment:
can be a warning

Reply:
it isn't dangerous or something like this. I used warning for cases where data can be lost.